### ISSUE TITLE
Error: viewer certificate conflicts

### ISSUE THREAD

**Issue by VSVS** [squidfunk/terraform-aws-cognito-auth#480]
Hey @squidfunk!

## Description
Im getting an error back that im not sure how to debug. After i run `$ terraform plan -out the.plan` i get these two errors:

```
Error: module.cognito-auth.module.web.aws_cloudfront_distribution._: "viewer_certificate.0.acm_certificate_arn": conflicts with viewer_certificate.0.cloudfront_default_certificate

Error: module.cognito-auth.module.web.aws_cloudfront_distribution._: "viewer_certificate.0.cloudfront_default_certificate": conflicts with viewer_certificate.0.acm_certificate_arn
```

my terraform code:

```
module "cognito-auth" {
    source  = "git@github.com:squidfunk/terraform-aws-cognito-auth.git"
    version = "0.3.4"

    # Required
    namespace                      = "myapp.mydomain.com"
    region                         = "${data.terraform_remote_state.region.name}"
    cognito_identity_pool_name     = "myapp mydomain com"
    cognito_identity_pool_provider = "login.mydomain.com"
    api_stage                      = "development"       

    # Optional: Default UI
    app_hosted_zone_id             = "${data.terraform_remote_state.provider.my_zone_id}"
    app_certificate_arn            = "${data.terraform_remote_state.region.us_east_2_myapp_cert_arn}"
    app_domain                     = "login.mydomain.com"
    app_origin                     = "admin.mydomain.com"

    # Optional: Email delivery
    ses_sender_address             = "rob@gmail.com"
}
```

### Expected behavior
Should create necessary resources to use this module

### Actual behavior

returns 2 errors.

### Steps to reproduce the bug

1. add module to my providers region
2. Configure files and folders
3. run `$ terraform plan -out the.plan`

### Package versions

* terraform-aws-cognito-auth: `0.3.4`
* terraform: `0.11.8`

### System information

* OS: 10.14

**Comment by squidfunk** [comment]
Hmm, strange, never encountered this problem. Does this also happen with the latest Terraform version? 0.11.8 seems pretty old. I will try to reproduce it, when I find the time.

**Comment by VSVS** [comment]
Thank you @squidfunk 

I tried this module with the latest terraform version .12.1 and noticed it was not compatible with the language changes used in version .12

I upgraded to the latest .11.14 and got the same error:

```
$ terraform -v
Terraform v0.11.14
+ provider.aws v2.0.0
+ provider.external v1.1.2
+ provider.null v2.1.2
+ provider.template v2.1.2

Your version of Terraform is out of date! The latest version
is 0.12.1. You can update by downloading from www.terraform.io/downloads.html
$ terraform plan -out the.plan

Error: module.cognito-auth.module.web.aws_cloudfront_distribution._: "viewer_certificate.0.acm_certificate_arn": conflicts with viewer_certificate.0.cloudfront_default_certificate

Error: module.cognito-auth.module.web.aws_cloudfront_distribution._: "viewer_certificate.0.cloudfront_default_certificate": conflicts with viewer_certificate.0.acm_certificate_arn
```

**Comment by squidfunk** [comment]
Okay, thanks for testing. I will try to upgrade the module to be compatible with Terraform 0.12. 

Regarding your problem: did you create your certificate in `us-east-1`? CloudFront demands that certificates are created in this region, otherwise it will not accept the certificate, see [this link](https://docs.aws.amazon.com/acm/latest/userguide/acm-regions.html).

**Comment by squidfunk** [comment]
Okay, so I took the chance to directly upgrade to the latest Terraform version. The code is currently on the `fix/upgrade-terraform` branch. Here is what I did:

– Destroyed the current demo deployment with `terraform destroy`
– Upgraded to Terraform 0.12
– Made all necessary changes so it's compatible with Terraform 0.12
– Deployed everything again with `terraform apply`

I haven't encountered the problem you described, so either it's gone now, or (what's more likely) it's an issue with your certificate, as described in my last comment. I will shortly issue a new release to ensure compatibility with Terraform 0.12.
